LanguageFeatures/Records/type_inference_A01_t01.dart

```diff
@@ -0,0 +1,34 @@
+// Copyright (c) 2022, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion Type inference and promotion flows through records in much the 
+/// same way it does for instances of generic classes (which are covariant in 
+/// Dart just like record fields are) and collection literals.
+///
+/// @description Checks type inference for records.
+/// @author sgrekhov22@gmail.com
+
+// SharedOptions=--enable-experiment=records
+
+import "../../Utils/static_type_helper.dart";
+
+main() {
+  var r1 = (42, "");
+  r1.expectStaticType<Exactly<(int, String)>>();
+
+  final r2 = (null, 3.14);
+  r2.expectStaticType<Exactly<(Null, double)>>();
```

Comment:
This is a very interesting case! We have the following:

```dart
void main() {
  var x = null;
  String s = x; // No compile-time error.
}
```

The point is that a variable of type `Null` isn't very useful, so type inference chooses `dynamic` as the declared type when a variable initializer has the type `Null`.

But I can see that we get `Null` as an inferred type argument for normal instance creation inference:

```dart
class C<X, Y> {
  final X x;
  final Y y;
  C(this.x, this.y);
}

void main() {
  var c = C(null, 3.14);
  print(c.runtimeType); // 'C(Null, double)'.
}
```

With that in mind I think it's OK to expect that `null` gives rise to an inferred record field type of `Null`.